compiler/src/ct_checker_forward.ml

```diff
@@ -172,7 +172,7 @@ let pp_kind fmt k =
   if k = Flexible then Format.fprintf fmt "#%s " (string_of_lvl_kind k)
 
 let pp_arg fmt (x, (k, lvl)) =
-  Format.fprintf fmt "%a%a %a" pp_kind k  Lvl.pp lvl (Printer.pp_var ~debug:false) x
+  Format.fprintf fmt "%a%a %a" pp_kind k  Lvl.pp lvl (Printer.pp_var ~debug:!Glob_options.debug) x
```

Comment:
This seems to be not the correct fix.
Debug may print many other infos that are usually used 
to debug the compiler. 
For doit, I am agree that it is important to print the info
(it is done after lower spill), name clashes can occur. But either replace this by just a true or a boolean 
depending on the pass.

Reply:
I see, if this is set depending on the pass then the pass would need to be passed through to the checker. I think this will be quite cumbersome.

Reply:
Indeed, fixing this is non trivial and mostly unrelated to this PR. Therefore please:
1. open an issue
2. drop this commit from this PR.

Reply:
Done: #810.